src/vectorDBManager.ts

```diff
@@ -79,56 +109,105 @@ class VectorDBManager {
     }
   }
 
-  public static async getMemoryVectors(docHash: string): Promise<MemoryVector[] | undefined> {
+  public static async loadFile(noteFile: NoteFile, embeddingsAPI: Embeddings) {
     if (!this.db) throw new Error("DB not initialized");
+    const textSplitter = new RecursiveCharacterTextSplitter({ chunkSize: 5000 })
+    const splitDocument = await textSplitter.createDocuments([noteFile.content])
+    const docVectors = await embeddingsAPI.embedDocuments(splitDocument.map((doc) => doc.pageContent))
+    const memoryVectors = docVectors.map((docVector, i) => (
+      {
+        content: splitDocument[i].pageContent,
+        metadata: {
+          ...noteFile.metadata,
+          title: noteFile.basename,
+          path: noteFile.path,
+
+        },
+        embedding: docVector,
+      }))
+    const docHash = VectorDBManager.getDocumentHash(noteFile.path);
+
+    const serializedMemoryVectors = JSON.stringify(memoryVectors);
     try {
-      const doc: VectorStoreDocument = await this.db.get(docHash);
-      if (doc && doc.memory_vectors) {
-        return JSON.parse(doc.memory_vectors);
-      }
+      // Attempt to fetch the existing document, if it exists.
+      const existingDoc = await this.db.get(docHash).catch(err => null);
+
+      // Prepare the document to be saved.
+      const docToSave = {
+        _id: docHash,
+        memory_vectors: serializedMemoryVectors,
+        file_mtime: noteFile.mtime,
+        created_at: Date.now(),
+        _rev: existingDoc?._rev // Add the current revision if the document exists.
+      };
+
+      // Save the document.
+      await this.db.put(docToSave);
     } catch (err) {
-      console.log("No vectors found in VectorDB for dochash:", docHash);
+      console.error("Error storing vectors in VectorDB:", err);
     }
   }
 
-  public static async removeOldDocuments(ttl: number): Promise<void> {
+  public static async getNoteFiles(): Promise<NoteFile[]> {
     if (!this.db) throw new Error("DB not initialized");
-
     try {
-      const thresholdTime = Date.now() - ttl;
-
-      // Fetch all documents from the database
-      const allDocsResponse = await this.db.allDocs<{ created_at: number }>({ include_docs: true });
-
-      // Filter out the documents older than 2 weeks
-      const oldDocs = allDocsResponse.rows.filter(row => {
-          // Assert the doc type
-          const doc = row.doc as VectorStoreDocument;
-          return doc && doc.created_at < thresholdTime;
+      const allDocsResponse = await this.db.allDocs<VectorStoreDocument>({ include_docs: true });
+      const allDocs = allDocsResponse.rows.map(row => row.doc as VectorStoreDocument);
+      const noteFiles = allDocs.map(doc => {
+        const memoryVectors = JSON.parse(doc.memory_vectors) as MemoryVector[];
+        const noteFile: NoteFile = {
+          path: memoryVectors[0].metadata.path,
```

Comment:
From the `loadFile` function above it looks like a pretty solid guarantee that it would always be not empty but it might be worth adding a bit of "defensive coding" in there just in case 🤔